src/Compilers/CSharp/Portable/CodeGen/CodeGenerator.cs

```diff
@@ -133,7 +167,7 @@ private LocalDefinition LazyReturnTemp
                             constraints: LocalSlotConstraints.None,
                             isDynamic: false,
                             dynamicTransformFlags: ImmutableArray<TypedConstant>.Empty,
-                            isSlotReusable: localSymbol.SynthesizedKind.IsSlotReusable(_optimizations));
+                            isSlotReusable: _ilEmitStyle == ILEmitStyle.Release);
```

Comment:
We already checked that _ilEmitStyle == ILEmitStyle.Debug, should we simply pass false here?